ToolBox/Benzinga/BenzingaDataConverter.cs

```diff
@@ -0,0 +1,346 @@
+                return false;
+            }
+
+            WriteToFile(Process(newsArticles), date);
+            return true;
+        }
+
+        /// <summary>
+        /// Process the data to BenzingaNews instances
+        /// </summary>
+        /// <param name="dateDirectory">Directory for a given date, e.g. ./root/2018/12/27/</param>
+        /// <returns></returns>
+        private IEnumerable<BenzingaNews> Process(DirectoryInfo dateDirectory)
+        {
+            foreach (var publication in dateDirectory.GetFiles("*.xml", SearchOption.TopDirectoryOnly))
+            {
+                var document = new XmlDocument();
+                document.LoadXml(File.ReadAllText(publication.FullName));
+
+                var item = JsonConvert.DeserializeObject<JObject>(JsonConvert.SerializeXmlNode(document))["rss"]["channel"]["item"];
+                // Benzinga complies with RSS standards and uses RFC 822 for DateTime
+                var publicationDate = Parse.DateTime(item["pubDate"].ToString());
+
+                var tickers = new List<JToken>();
+
+                // Only process articles that contain tickers to disk
+                if (item["bz:ticker"] == null)
+                {
+                    continue;
+                }
+
+                // Check for only a single ticker before we try to iterate
+                if (item["bz:ticker"].Type == JTokenType.Object)
+                {
+                    tickers.Add(item["bz:ticker"]);
+                }
+                else
+                {
+                    foreach (var tickerData in JArray.Parse(item["bz:ticker"].ToString()))
+                    {
+                        tickers.Add(tickerData);
+                    }
+                }
+
+                var instance = JsonConvert.DeserializeObject<BenzingaNews>(item.ToString());
+
+                // Strip all HTML tags from the article, then convert HTML entities to their string representation
+                // e.g. "<html><p>Apple&#39;s Earnings</p></html>" would become "Apple's Earnings"
+                instance.Contents = WebUtility.HtmlDecode(Regex.Replace(instance.Contents, @"<[^>]*>", " "));
+                instance.Symbols = new List<BenzingaSymbolData>();
+                instance.Metadata = new BenzingaMetadata()
+                {
+                    IsPro = item["bz:type"]["@pro"].ToString() == "1",
+                    FirstRun = item["bz:type"]["@pro"].ToString() == "1",
+                    Kind = item["bz:type"]["#text"].ToString()
+                };
+
+                foreach (var ticker in tickers)
+                {
+                    var sentiment = ticker["@sentiment"] == null ? (decimal?)null : Parse.Decimal(ticker["@sentiment"].ToString());
+                    var exchange = ticker["@exchange"] == null ? string.Empty : ticker["@exchange"].ToString();
+
+                    if (ticker["#text"] == null)
+                    {
+                        continue;
```

Comment:
it's good style to place guard clauses at the very top of the scope. In this case, there's no value in parsing `sentiment` and `exchange` just to get to a bad `#text` value to skip the entire `ticker` -- but again -- I think most of the code in this loop collapses -- I recommend _briefly_ reviewing those docs, reading this code it looks like you're going against the grain of the library.

I'm gonna share something else here totally unrelated only because it's one of the coolest features of the library `JToken` side -- I actually went and found an [SO question](https://stackoverflow.com/a/29782523/1582922) I could answer using the feature so I would never lose it!! Basically you can provide a string expression, such as "symbols[*].ticker" which would work on an object like:
``` js
{
  symbols: [
   { ticker: 'ticker-1' },
   { ticker: 'ticker-2' },
   { ticker: 'ticker-3' },
   { not-ticker: 'ticker-1' }
  ]
}
```

and it would happily ignore the `not-ticker` -- pretty nice feature when working using the JToken library. _That_ being said, in general, the JToken side of the house is typically used when you don't have the contracts coded up (custom class w/ `[JsonProperty]` attributes.